plugins/modem-manager/fu-mm-device.c

```diff
@@ -1605,6 +1653,76 @@ fu_mm_device_write_firmware_fdl(FuDevice *device, GBytes *fw, FuProgress *progre
 					      fw,
 					      error);
 }
+
+static gboolean
+fu_mm_device_write_firmware_dfota(FuDevice *device,
+				  GBytes *fw,
+				  FuProgress *progress,
+				  GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autofree gchar *upload_cmd = NULL;
+
+	locker = fu_device_locker_new_full(device,
+					   (FuDeviceLockerFunc)fu_mm_device_dfota_open,
+					   (FuDeviceLockerFunc)fu_mm_device_dfota_close,
+					   error);
+
+	if (locker == NULL)
+		return FALSE;
+	if (!fu_mm_device_at_cmd(self, "AT+QFLST=?", TRUE, error)) {
+		g_prefix_error(error, "listing files not supported: ");
+		return FALSE;
+	}
+	/* if listing firmware file does not fail, there is an old firmware file to remove */
+	if (fu_mm_device_at_cmd(self,
+				"AT+QFLST=\"UFS:" FU_DFOTA_UPDATER_FILENAME "\"",
+				TRUE,
+				NULL)) {
```

Comment:
I don't think you should discard the error here, instead use an error_local and output to the debug log what you found.